lib/std/io/reader.zig

```diff
@@ -101,6 +96,35 @@ pub fn Reader(
             return array_list.toOwnedSlice();
         }
 
+        /// Replaces the `std.ArrayList` contents by reading from the stream until `delimiter` or end-of-stream is found.
+        /// Does not include the delimiter in the result.
+        /// If the `std.ArrayList` length would exceed `max_size`, `error.StreamTooLong` is returned and the
+        /// `std.ArrayList` is populated with `max_size` bytes from the stream.
+        pub fn readUntilDelimiterOrEofArrayList(
+            self: Self,
+            array_list: *std.ArrayList(u8),
+            delimiter: u8,
+            max_size: usize,
+        ) !void {
+            array_list.shrink(0);
```

Comment:
I noticed that too. Perhaps this is something that should be revisited and discussed in a future PR? I was also thinking that maybe `shrinkRetainingCapacity` would be better in general, so that we don't perform unnecessary frees and allocs.

Reply:
I totally agree

Reply:
Yea that bit did creepy me out, but as `readUntilDelimiterArrayList` already did it I left it in.

A second having the functions just straight append without any shrinking.

What if you want to call this function multiple times on different readers to collect into a single array list?